Services/Dashboard/Block/classes/class.ilDashboardBlockGUI.php

```diff
@@ -0,0 +1,683 @@
+        );
+
+        $this->initAndShow();
+    }
+
+    public function changePDItemPresentationObject(): void
+    {
+        $this->viewSettings->storeActorPresentationMode(
+            ilUtil::stripSlashes((string) ($this->http->request()->getQueryParams()['presentation'] ?? ''))
+        );
+        $this->initAndShow();
+    }
+
+    protected function cancel(): void
+    {
+        $this->ctrl->returnToParent($this);
+    }
+
+    protected function returnToContext(): void
+    {
+        if ($this->http->request()->getQueryParams()['manage'] ?? false) {
+            $this->ctrl->redirect($this, 'manage');
+        }
+        $this->ctrl->redirectByClass('ildashboardgui', 'show');
+    }
+
+    /**
+     * @return array<string, ilBlockDTO[]>
+     */
+    public function getItemGroups(): array
+    {
+        switch ($this->viewSettings->getEffectiveSortingMode()) {
+            case ilPDSelectedItemsBlockConstants::SORT_BY_ALPHABET:
+                $data = $this->getData();
+                $data = array_merge(...array_values($data));
+                $data = $this->sortByTitle($data);
+                return ['' => $data];
+            case ilPDSelectedItemsBlockConstants::SORT_BY_START_DATE:
+                return $this->groupItemsByStartDate();
+            case ilPDSelectedItemsBlockConstants::SORT_BY_TYPE:
+                return $this->groupItemsByType();
+            case ilPDSelectedItemsBlockConstants::SORT_BY_LOCATION:
+            default:
+                return $this->groupItemsByLocation();
+        }
+    }
+
+    #[NoReturn]
+    public function removeFromDeskRoundtripObject(): void
+    {
+        $page = '';
+        if ($this->http->wrapper()->query()->has('page')) {
+            $page = $this->http->wrapper()->query()->retrieve('page', $this->refinery->kindlyTo()->string());
+        }
+
+        if ($this->http->wrapper()->query()->has('replaceSignal')) {
+            $signalId = $this->http->wrapper()->query()->retrieve('replaceSignal', $this->refinery->kindlyTo()->string());
+            $replace_signal = new ReplaceSignal($signalId);
+        }
+
+        switch ($page) {
+            case 'manage':
+                $modal = $this->ui->factory()->modal()->roundtrip(
+                    $this->getRemoveMultipleActionText(),
+                    $this->ui->factory()->legacy($this->manage($replace_signal ?? null))
+                );
+                $modal = $modal->withAdditionalOnLoadCode(function ($id) {
+                    return "
+                    $('#$id').attr('data-modal-name', 'remove_modal_view_" . $this->viewSettings->getCurrentView() . "');
+                    ";
+                });
+                break;
+            case 'confirm':
+            default:
+                $modal = $this->ui->factory()->legacy($this->confirmRemoveObject());
+        }
+        echo $this->ui->renderer()->renderAsync($modal);
+        exit;
```

Comment:
Please use `$DIC->http()` to send a response and exit (with `$DIC->http()->close()` etc).